EventListeners.js

```diff
@@ -0,0 +1,53 @@
+// Why does this bring an error?
+// import { GAME_MODE } from "./constants";
+
+
+// move basket when left or right key is pressed
+document.addEventListener('keydown', (e) => {
+    if (e.key === 'ArrowLeft') {
+        console.log('Left key was pressed')
+        // moveBasketLeft(state, 10)
+    }; 
+    
+    if (e.key === 'ArrowRight') {
+        console.log('Right key was pressed')
+        // moveBasketRight(state, 10)
+    };
+})
+
+// move basket when mouse is used
+const playArea = document.querySelector('.play-area');
+
+playArea.addEventListener('mousemove', (e) => {
+    let basketCenter = e.clientX - ( basket.width / 2 )
+    let xPos = basketCenter + 'px';
+    basketDiv.style.left = xPos;
+    // setBasketPos(state, xPos) mutator
+
+
+})
+
+// listen when p/enter/space/esc is pressed
+document.addEventListener('keydown', (e) => {
+    if (e.key === 'p' || 
+        e.key === ' ' || 
+        e.key === 'Escape' || 
+        e.key === 'Pause') {
+            // setGameMode(state, 'pause')
+    }
+})
+
+// Listen for when enter is pressed when game is paused, to resume game
+document.addEventListener('keydown', (e) => {
+    if (GAME_MODE === 'paused' && e.key === 'Enter') {
```

Comment:
This logic should be up in the previous event handler, as it's the same `keydown` event on the same `document` element. In there, if the pause key is pressed, you should check the current state and set the new state accordingly. So if the current is 'paused', the new should be 'running' etc

Reply:
I was wondering about that when I was first writing it but thought it would be better to have them separate. Thanks for clearing that up for me. Updated